katz/src/main/kotlin/katz/data/Validated.kt

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (C) 2017 The Katz Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package katz
+
+typealias ValidatedNel<E, A> = Validated<NonEmptyList<E>, A>
+
+sealed class Validated<out E, out A> : HK2<Validated.F, E, A> {
```

Comment:
create `typealias ValidatedKind<E,A> = HK2<Validated.F, E, A>` it will need to improve redeability of ValidadtedMonad